### PR TITLE
Support packed nonce

### PR DESCRIPTION
**Pull request by IIIIllllIIIIllllIIIIllllIIIIllllIIIIll**
This PR allows dependency nonces to be treated in a "packed" format (https://gist.github.com/IIIIllllIIIIllllIIIIllllIIIIllllIIIIll/af044ce89cdd6839453afcffa6c43302).

### CODE REVIEW

File: contracts/NonceRegistry.sol [hunk 1]
```diff
@@ -33,28 +33,81 @@ contract NonceRegistry {
...
-    return state.nonce == nonce;
+    require(
+      state.nonce == expectedNonce,
+      "Nonce is not equal to expectedNonce"
+    );
+    return true;
   }
 
-  /// @notice Set a nonce in the mapping and triggers the timeout period to begin
+  /// "Get First N Bits" from
```
Comment (snario) [inline contracts/NonceRegistry.sol]: Can you stick with the natstring format

File: test/unit/nonceRegistry.spec.ts [hunk 2]
```diff
@@ -17,6 +27,28 @@ contract("NonceRegistry", accounts => {
...
+    );
+  });
+
+  it("getFirstNBits works for 9", async () => {
+    const ret = await registry.functions.getFirstNBits(
+      "0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff",
+      9
+    );
+    // 0x8 == 0b1000
+    console.log(ret);
```
Comment (snario) [inline test/unit/nonceRegistry.spec.ts]: Shouldn't be included

**Comment by IIIIllllIIIIllllIIIIllllIIIIllllIIIIll** [comment]
When was nonce finalization without a timeout being used?

**Comment by armaniferrante** [comment]
@IIIIllllIIIIllllIIIIllllIIIIllllIIIIll I don't think nonce finalization without a timeout was being used anywhere. However, to get a POC of #59 to work, we added it to the protocols just so that the multisend transaction would go through on-chain. 

@snario is this what you were referring to?

**Comment by IIIIllllIIIIllllIIIIllllIIIIllllIIIIll** [comment]
closed by accident

**Comment by IIIIllllIIIIllllIIIIllllIIIIllllIIIIll** [comment]
If we added the finalization as a hack with a known reason, let's revert the removal from this PR and remove it / discuss its removal separately. The purpose of a dependency nonce implies that its unanimous finalization is not needed in an install and uninstall protocol.

**Pull request merged**